plan/refiner.go

```diff
@@ -65,7 +65,7 @@ func buildIndexRange(sc *variable.StatementContext, p *PhysicalIndexScan) error
 		}
 	}
 
-	if p.accessInAndEqCount+1 < len(p.Index.Columns) {
+	if len(p.Ranges[0].LowVal) < len(p.Index.Columns) {
 		for _, ran := range p.Ranges {
 			if ran.HighExclude || ran.LowExclude {
```

Comment:
1. Non-exclude needn't to change
2. we cannot make sure that, may be still less than the len of index cols.